mir-module/src/main/resources/xsl/mods2marcxml.xsl

```diff
@@ -0,0 +1,2820 @@
+					<xsl:when test=".='law report or digest'">w</xsl:when>
+					<xsl:when test=".='legal article'">g</xsl:when>
+					<xsl:when test=".='legal case and case notes'">v</xsl:when>
+					<xsl:when test=".='legislation'">l</xsl:when>
+					<xsl:when test=".='patent'">j</xsl:when>
+					<xsl:when test=".='programmed text'">p</xsl:when>
+					<xsl:when test=".='review'">o</xsl:when>
+					<xsl:when test=".='statistics'">s</xsl:when>
+					<xsl:when test=".='survey of literature'">n</xsl:when>
+					<xsl:when test=".='technical report'">t</xsl:when>
+					<xsl:when test=".='theses'">m</xsl:when>
+					<xsl:when test=".='treaty'">z</xsl:when>
+				</xsl:choose>
+			</xsl:for-each>
+		</xsl:variable>
+		<xsl:call-template name="makeSize">
+			<xsl:with-param name="string" select="$chars"/>
+			<xsl:with-param name="length" select="4"/>
+		</xsl:call-template>
+	</xsl:template>
+
+	<xsl:template name="controlField008-30-31">
+		<xsl:variable name="chars">
+			<xsl:for-each select="mods:genre[@authority='marc']">
+				<xsl:choose>
+					<xsl:when test=".='biography'">b</xsl:when>
+					<xsl:when test=".='conference publication'">c</xsl:when>
+					<xsl:when test=".='drama'">d</xsl:when>
+					<xsl:when test=".='essay'">e</xsl:when>
+					<xsl:when test=".='fiction'">f</xsl:when>
+					<xsl:when test=".='folktale'">o</xsl:when>
+					<xsl:when test=".='history'">h</xsl:when>
+					<xsl:when test=".='humor, satire'">k</xsl:when>
+					<xsl:when test=".='instruction'">i</xsl:when>
+					<xsl:when test=".='interview'">t</xsl:when>
+					<xsl:when test=".='language instruction'">j</xsl:when>
+					<xsl:when test=".='memoir'">m</xsl:when>
+					<xsl:when test=".='rehersal'">r</xsl:when>
+					<xsl:when test=".='reporting'">g</xsl:when>
+					<xsl:when test=".='sound'">s</xsl:when>
+					<xsl:when test=".='speech'">l</xsl:when>
+				</xsl:choose>
+			</xsl:for-each>
+		</xsl:variable>
+		<xsl:call-template name="makeSize">
+			<xsl:with-param name="string" select="$chars"/>
+			<xsl:with-param name="length" select="2"/>
+		</xsl:call-template>
+	</xsl:template>
+
+	<xsl:template name="makeSize">
+		<xsl:param name="string"/>
+		<xsl:param name="length"/>
+		<xsl:variable name="nstring" select="normalize-space($string)"/>
+		<xsl:variable name="nstringlength" select="string-length($nstring)"/>
+		<xsl:choose>
+			<xsl:when test="$nstringlength&gt;$length">
+				<xsl:value-of select="substring($nstring,1,$length)"/>
+			</xsl:when>
+			<xsl:when test="$nstringlength&lt;$length">
+				<xsl:value-of select="$nstring"/>
+				<xsl:call-template name="buildSpaces">
+					<xsl:with-param name="spaces" select="$length - $nstringlength"/>
+					<xsl:with-param name="char">|</xsl:with-param>
+				</xsl:call-template>
+			</xsl:when>
+			<xsl:otherwise>
+				<xsl:value-of select="$nstring"/>
+			</xsl:otherwise>
+		</xsl:choose>
+	</xsl:template>
+
+	<xsl:template match="mods:mods">
+		<marc:record>
+			<marc:leader>
+				<!-- 00-04 -->
+				<xsl:text>     </xsl:text>
```

Comment:
these 4 whitespaces seems to be ignored in XML output
<marc:leader>nam  22     uu 4500</marc:leader>

Reply:
works for me, please see source of the output not the browser output: https://dbttest.thulb.uni-jena.de/dbt-master/receive/dbt_mods_00032675?XSL.Transformer=marcxml